data/actions/scripts/other/doors.lua

```diff
@@ -1,3 +1,51 @@
+local positions = {
+	{x = 1, y = 0}, -- east
+	{x = -1, y = 1}, -- south
+	{x = -1, y = -1}, -- west
+	{x = 1, y = -1} -- north
+}
+
+local function moveToAptPos(creature, round)
```

Comment:
I don't understand this function name. Could you make it more obvious as to what it does? What is Apt?

Reply:
You guessed correctly, Apt is another word for appropriate.
I'll argue that it shouldn't be changed as it's not an abbreviation, but it's own word.

Reply:
This isn't a clear function name, regardless of the word chosen. Maybe add a description of what and how it does what it does?

Reply:
What about the function name `pushFromDoor` or `pushCreatureFromDoor`?
Since this is only used in doors.lua, and this is strictly speaking, what the function does?
"AptPos" is a bad abbrevation, and "AppropriatePosition" is ambigious.

Reply:
Apt is not an abbreviation, it has a meaning closer to "possible" rather than "appropriate". "Apt" is an apt word here, but maybe not appropriate :P

However I would rename it to `findKickPosition` and return the position instead of moving the player, and move inside `onUse` to avoid a function with multiple responsibilities.

Reply:
To be precise apt is a homonym so the meaning depends on the context in which it is used, either way I agree that it is ambiguous.

How about findPushPosition? Imo push fits better than kick when a creature is being relocated by a closing door.

To be sure, by "move inside onUse" do you mean I should nest findPushPosition inside onUse?
And thanks I didn't know about Curly's law before.